src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +89,55 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
-            // Description
+            // DisplayFormatString
+            if (displayFormatAttribute != null)
+            {
+                displayMetadata.DisplayFormatString = displayFormatAttribute.DataFormatString;
+            }
+
+            // DisplayDescription
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
+                    !string.IsNullOrEmpty(displayAttribute.Name) &&
+                    displayAttribute.ResourceType == null)
```

Comment:
Isn't that covered by [these](https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs#L158) tests? Or am I misunderstanding what you want tested?